docs/design/resource-states.md

```diff
@@ -0,0 +1,222 @@
+
+> In general, condition values may change back and forth, but some condition transitions may be monotonic, depending on the resource and condition type. However, conditions are observations and not, themselves, state machines, nor do we define comprehensive state machines for objects, nor behaviors associated with state transitions. The system is level-based rather than edge-triggered, and should assume an Open World.
+
+> Controllers should apply their conditions to a resource the first time they visit the resource, even if the status is Unknown. This allows other components in the system to know that the condition exists and the controller is making progress on reconciling that resource.
+
+These from [standardizing conditions](https://github.com/kubernetes/enhancements/tree/master/keps/sig-api-machinery/1623-standardize-conditions):
+
+> reason is required and must not be empty. The actor setting the value should always describe why the condition is the way it is, even if that value is "unknown unknowns". No other actor has the information to make a better choice.
+
+Some other best practices on [Kubernetes design principles](https://github.com/kubernetes/community/blob/4c9ef2d/contributors/design-proposals/architecture/principles.md):
+
+> Object status must be 100% reconstructable by observation. Any history kept must be just an optimization and not required for correct operation.
+
+> Do not define comprehensive state machines for objects with behaviors associated with state transitions and/or "assumed" states that cannot be ascertained by observation.
+
+### Details
+All resource's `Status`'s will have a top level `Conditions` field which is a collection of type `Condition`. This collection supports extension (through conditions with different type names).
+Initially we will expose a single `Ready` condition across all resources, representing the high level availability of the resource and its readiness for use.
+
+The structure of the `Condition` is based on the [reccomended shape of conditions KEP](https://github.com/kubernetes/enhancements/tree/master/keps/sig-api-machinery/1623-standardize-conditions) as well as the [Cluster API conditions propsal](https://github.com/kubernetes-sigs/cluster-api/blob/master/docs/proposals/20200506-conditions.md) and is as follows:
+
+```go
+// ConditionSeverity expresses the severity of a Condition Type failing.
+type ConditionSeverity string
+
+const (
+  // ConditionSeverityError specifies that a failure of a condition type
+  // should be viewed as an error.
```

Comment:
Is this where we should be explicit that an error will require external intervention to address? (vs a warning that we might be able to resolve automatically.)

Reply:
Yes. I've updated the comments here quite a bit based on an actual copy of this code I have in a branch. It should be clearer now.